packages/react/src/components/Masthead/MastheadSearch.js

```diff
@@ -169,12 +169,14 @@ const MastheadSearch = ({
     /**
      * Close search when click detected outside of component.
      * This is necessary otherwise search stays open even when
-     * elements other than the close button are clicked.
+     * elements other than the close button and the
+     * profile button are clicked.
      *
-     * @param {*} event Click event outside search component
+     * @param {*} event Click event outside masthead component
      */
     const handleClickOutside = event => {
-      if (ref.current && !ref.current.contains(event.target)) {
+      let mastheadRef = ref.current.parentNode.parentNode;
```

Comment:
Would you try `ref.current?.closest('.some--css-class')` to avoid reliance to the specific DOM structure?